src/main/kotlin/me/zeroeightsix/kami/feature/command/NbtCommand.kt

```diff
@@ -0,0 +1,221 @@
+
+            literal("self") {
+                does {
+                    open(mc.player?.toTag(CompoundTag()) ?: return@does 1)
+                    0
+                }
+            }
+
+            literal("hand") {
+                does {
+                    val stack = mc.player?.mainHandStack
+                    if (stack?.isEmpty == true) {
+                        Error.NO_ITEM.send(it.source)
+                        return@does 1
+                    } else {
+                        val tag = stack?.tag
+                            ?: run { Error.NO_ITEM_NBT.send(it.source); return@does 1 }
+                        open(tag)
+                    }
+                    0
+                }
+            }
+        }
+    }
+
+    private fun open(tag: Tag) {
+        screen.tag = tag
+        KamiMod.EVENT_BUS.subscribe(opener)
+    }
+
+    private enum class Error(val message: Text) {
+        NO_TARGET(text(Formatting.RED) { +"No Target found!" }),
+        NO_BLOCK_ENTITY(text(Formatting.RED) { +"This Block is not a BlockEntity!" }),
+        NO_ITEM(text(Formatting.RED) { +"You must hold an item!" }),
+        NO_ITEM_NBT(text(Formatting.RED) { +"The item you are holding has no NBT!" });
+
+        fun send(source: CommandSource) {
+            source replyWith message as MutableText
+        }
+    }
+}
+
+class NbtScreen(
+    var tag: Tag,
+    val defaultColor: Vec4 = Vec4(1f, 1f, 1f, 1f)
+) : InteractableScreen(text(null, "Kami NBT")),
+    widgetsText,
+    widgetsMain,
+    widgetsTrees,
+    contentRegion,
+    childWindows,
+    windowsUtilities,
+    itemWidgetsUtilities,
+    columns,
+    cursorLayout
+// this comment is here to prevent intelli'ntJ from formatting this against the kotlin conventions
+{
+    override fun render(matrices: MatrixStack?, mouseX: Int, mouseY: Int, delta: Float) {
+        super.render(matrices, mouseX, mouseY, delta)
+
+        KamiHud.frame(matrices!!) {
+            this()
+        }
+    }
+
+    operator fun invoke() {
+        dsl.window("NBT") {
+            if (button("Copy", Vec2(windowContentRegionWidth, 0)))
+                copyTagToClipboard(tag)
+            beginChild("##nbtPane", border = true)
+            columns(2)
+            showTree("NBT", tag)
+            columns(1)
+            endChild()
+        }
+    }
+
+    private fun showTree(tagName: String, tag: Tag, appendix: Int = 0) {
+        alignTextToFramePadding()
+        var curApp = appendix
+        if (treeNodeEx("$tagName##$curApp", TreeNodeFlag.NoTreePushOnOpen.i)) {
+            nextColumn()
+            nextColumn()
+            treePush()
+            when (tag) {
+                is CompoundTag -> {
+                    for (key in tag.keys)
+                        tag[key]?.let {
+                            if (tagIsDeep(it))
+                                showTree(key, it, ++curApp)
+                            else
+                                dsl.withStyleColor(Col.Text, tagColor(it)) {
```

Comment:
```suggestion
                                withStyleColor(Col.Text, tagColor(it)) {
```
Add an import for `withStyleColor`.